src/cocotb/_scheduler.py

```diff
@@ -251,29 +256,23 @@ def __init__(self, test_complete_cb: Callable[[], None]) -> None:
             SimHandleBase, Tuple[Callable[..., None], Sequence[Any]]
         ] = OrderedDict()
 
-        self._pending_tasks = []
-        self._pending_triggers = []
+        self._pending_tasks: OrderedDict[Task[Any], _outcomes.Outcome] = OrderedDict()
         self._pending_threads = []
         self._pending_events = []  # Events we need to call set on once we've unwound
-        self._scheduling = []
 
         self._terminate = False
         self._test = None
         self._main_thread = threading.current_thread()
 
         self._current_task = None
 
-        self._is_reacting = False
-
         self._write_task = None
         self._writes_pending = Event()
 
     async def _do_writes(self):
         """An internal task that performs pending writes"""
         while True:
             await self._writes_pending.wait()
-            if self._mode != Scheduler._MODE_NORMAL:
-                await self._next_time_step
```

Comment:
This is not possible since `_schedule_write` can't be called except in NORMAL, and set is only called there.